app/Http/Controllers/BeatmapDiscussionPostsController.php

```diff
@@ -112,60 +111,31 @@ public function store()
     {
         /** @var User $user */
         $user = auth()->user();
-        $params = request()->all();
-        $discussion = $this->prepareDiscussion($params);
+        $request = request()->all();
 
-        if (!$discussion->exists) {
-            priv_check('BeatmapDiscussionStore', $discussion)->ensureCan();
-        }
+        $discussionId = get_int($request['beatmap_discussion_id'] ?? null);
+        $beatmapsetId = get_int($request['beatmapset_id'] ?? null);
 
-        $postParams = get_params($params, 'beatmap_discussion_post', ['message']);
-        $postParams['user_id'] = $user->getKey();
-        $post = new BeatmapDiscussionPost($postParams);
-        $post->beatmapDiscussion()->associate($discussion);
-
-        priv_check('BeatmapDiscussionPostStore', $post)->ensureCan();
-
-        $event = BeatmapsetEvent::getBeatmapsetEventType($discussion, $user);
-        $notifyQualifiedProblem = $discussion->shouldNotifyQualifiedProblem($event);
-
-        $posts = $discussion->getConnection()->transaction(function () use ($discussion, $event, $post, $user) {
-            $discussion->saveOrExplode();
-
-            // done here since discussion may or may not previously exist
-            $post->beatmap_discussion_id = $discussion->getKey();
-            $post->saveOrExplode();
-            $newPosts = [$post];
-
-            switch ($event) {
-                case BeatmapsetEvent::ISSUE_REOPEN:
-                case BeatmapsetEvent::ISSUE_RESOLVE:
-                    $systemPost = BeatmapDiscussionPost::generateLogResolveChange($user, $discussion->resolved);
-                    $systemPost->beatmap_discussion_id = $discussion->getKey();
-                    $systemPost->saveOrExplode();
-                    BeatmapsetEvent::log($event, $user, $post)->saveOrExplode();
-                    $newPosts[] = $systemPost;
-                    break;
-
-                case BeatmapsetEvent::DISQUALIFY:
-                case BeatmapsetEvent::NOMINATION_RESET:
-                    $discussion->beatmapset->disqualifyOrResetNominations($user, $discussion);
-                    break;
-            }
+        if ($discussionId !== null) {
+            $discussion = BeatmapDiscussion::findOrFail($discussionId);
+            $resolve = get_bool($request['beatmap_discussion']['resolved'] ?? null);
+            $message = presence(get_string($request['beatmap_discussion_post']['message'] ?? null));
```

Comment:
inside the class just like `BeatmapsetDiscussionNew`? Seems weird one is dealt here and the other is in the class.